packages/api/cli/src/util/resolve-working-dir.ts

```diff
@@ -0,0 +1,25 @@
+import path from 'node:path';
+
+import fs from 'fs-extra';
+
+/**
+ * Resolves the directory in which to use
+ * @param dir - The directory specified by the user (can be relative or absolute)
+ * @param checkExisting - Checks if the directory already exists
+ * @returns
+ */
+export function resolveWorkingDir(dir: string, checkExisting = true): string {
+  let finalDir = process.cwd();
+  if (dir) {
+    if (path.isAbsolute(dir) && (!checkExisting || fs.existsSync(dir))) {
+      finalDir = dir;
+    } else {
+      const resolved = path.resolve(finalDir, dir);
+      if (!checkExisting || fs.existsSync(resolved)) {
+        finalDir = resolved;
+      }
+    }
+  }
+
+  return finalDir;
```

Comment:
Reduce nesting

```suggestion
  let finalDir = process.cwd();
  if (!dir) {
  	return finalDir;
  }
  
  const resolved = path.isAbsolute(dir)
  	? dir
  	: path.resolve(finalDir, dir);

  if (!checkExisting || fs.existsSync(resolved)) {
    return resolved;
  }
  
  return finalDir;
```

Reply:
https://github.com/electron/forge/pull/3851/commits/e38fd65094750e85c822b14e9871ac7214067656

Simplified it a bit more